### PR TITLE
Fix missing unique=True. Fix versioning changes

### PR DESCRIPTION
**Pull request by amagdas**
Sync to latest version of Eve.
Fix all regressions introduced.

### CODE REVIEW

**Comment by sivakuna-aap** [comment]
@amagdas can you please mention the ticket number or description of the prob you are trying address with PR?

File: server/apps/publish/archive_publish_tests.py [hunk 1]
```diff
@@ -323,8 +323,11 @@ def tearDown(self):
             os.remove(self.filename)
 
     def _init_article_versions(self):
+        resource_def = self.app.config['DOMAIN']['archive_versions']
+        version_id = versioned_id_field(resource_def)
+        print(version_id)
```
Comment (plamut) [inline server/apps/publish/archive_publish_tests.py]: Has this been put here intentionally or is it maybe a forgotten statement?

File: server/apps/publish/archive_publish_tests.py [hunk 2]
```diff
@@ -396,11 +399,14 @@ def _get_legal_archive_details(self, article_id, publishing_action=None):
         publish_queue_service = get_resource_service(LEGAL_PUBLISH_QUEUE_NAME)
 
         article = archive_service.find_one(_id=article_id, req=None)
-        article_versions = archive_versions_service.get(req=None, lookup={versioned_id_field(): article_id})
+        resource_def = self.app.config['DOMAIN'][LEGAL_ARCHIVE_VERSIONS_NAME]
+        version_id = versioned_id_field(resource_def)
+        print('Getting from legal archive versions: ', version_id, article_id)
```
Comment (plamut) [inline server/apps/publish/archive_publish_tests.py]: Same question here - forgotten or intended `print()` call?

File: server/apps/storage/amazon/amazon_media_storage_test.py [hunk 3]
```diff
@@ -1,24 +0,0 @@
...
-        app.config['AMAZON_S3_USE_HTTPS'] = True
-        app.config['AMAZON_SERVE_DIRECT_LINKS'] = True
-        self.assertEquals(AmazonMediaStorage(app).url_for_media(media_id),
-                          'https://AMAZON_CONTAINER_NAME.s3-us-east-1.amazonaws.com/%s' % (media_id))
-        app.config['AMAZON_S3_USE_HTTPS'] = False
-        self.assertEquals(AmazonMediaStorage(app).url_for_media(media_id),
-                          'http://AMAZON_CONTAINER_NAME.s3-us-east-1.amazonaws.com/%s' % (media_id))
-        app.config['AMAZON_REGION'] = 'eu-west-1'
-        self.assertEquals(AmazonMediaStorage(app).url_for_media(media_id),
-                          'http://AMAZON_CONTAINER_NAME.s3-eu-west-1.amazonaws.com/%s' % (media_id))
```
Comment (plamut) [inline server/apps/storage/amazon/amazon_media_storage_test.py]: Was this deleted or just moved somewhere else? Just curious.

**Comment by plamut** [comment]
So far so good IMO...

File: server/apps/storage/amazon/amazon_media_storage_test.py (hunk 3 above)
Comment (amagdas) [inline server/apps/storage/amazon/amazon_media_storage_test.py]: It's moved to core.

**Comment by mugurrus** [comment]
:+1:

**Pull request merged**